jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketChannel.java

```diff
@@ -262,7 +262,8 @@ public boolean isClosed()
     public void setWebSocketConnection(WebSocketConnection connection)
     {
         this.connection = connection;
-        getConnection().getEndPoint().setIdleTimeout(idleTimeout.toMillis());
+        if (idleTimeout != null)
+            getConnection().getEndPoint().setIdleTimeout(idleTimeout.toMillis());
```

Comment:
.. and then null the `idleTimeout` in a half hearted attempt to avoid duplicate state!

Reply:
@lachlan-roberts fix this one and I'll merge

Reply:
i already pushed this change, look to the line below